Terminal.Gui/View/Layout/Pos.cs

```diff
@@ -0,0 +1,694 @@
+    ///     subclass of Pos that is used. For example, PosAbsolute returns a fixed position, PosAnchorEnd returns a
+    ///     position that is anchored to the end of the layout, and so on.
+    /// </returns>
+    internal virtual int Anchor (int width) { return 0; }
+
+    /// <summary>
+    ///     Calculates and returns the position of a <see cref="View"/> object. It takes into account the dimension of the
+    ///     superview and the dimension of the view itself.
+    /// </summary>
+    /// <param name="superviewDimension">
+    ///     The dimension of the superview. This could be the width for x-coordinate calculation or the
+    ///     height for y-coordinate calculation.
+    /// </param>
+    /// <param name="dim">The dimension of the View. It could be the current width or height.</param>
+    /// <param name="us">The View that holds this Pos object.</param>
+    /// <param name="dimension">Width or Height</param>
+    /// <returns>
+    ///     The calculated position of the View. The way this position is calculated depends on the specific subclass of Pos
+    ///     that
+    ///     is used.
+    /// </returns>
+    internal virtual int Calculate (int superviewDimension, Dim dim, View us, Dimension dimension) { return Anchor (superviewDimension); }
+
+    /// <summary>
+    ///     Diagnostics API to determine if this Pos object references other views.
+    /// </summary>
+    /// <returns></returns>
+    internal virtual bool ReferencesOtherViews () { return false; }
+}
+
+/// <summary>
+///    Represents an absolute position in the layout. This is used to specify a fixed position in the layout.
+/// </summary>
+/// <remarks>
+///     <para>
+///     This is a low-level API that is typically used internally by the layout system. Use the various static
+///     methods on the <see cref="Pos"/> class to create <see cref="Pos"/> objects instead.
+///     </para>
+/// </remarks>
+/// <param name="position"></param>
+public class PosAbsolute (int position) : Pos
+{
+    /// <summary>
+    ///    The position of the <see cref="View"/> in the layout.
+    /// </summary>
+    public int Position { get; } = position;
+
+    /// <inheritdoc />
+    public override bool Equals (object other) { return other is PosAbsolute abs && abs.Position == Position; }
+
+    /// <inheritdoc />
+    public override int GetHashCode () { return Position.GetHashCode (); }
+
+    /// <inheritdoc />
+    public override string ToString () { return $"Absolute({Position})"; }
+
+    internal override int Anchor (int width) { return Position; }
+}
+
+/// <summary>
+///     Represents a position anchored to the end (right side or bottom).
+/// </summary>
+/// <remarks>
+///     <para>
+///     This is a low-level API that is typically used internally by the layout system. Use the various static
+///     methods on the <see cref="Pos"/> class to create <see cref="Pos"/> objects instead.
+///     </para>
+/// </remarks>
+public class PosAnchorEnd : Pos
```

Comment:
I don't think it should be in this PR, but combining all the AchnorXXX types into a single `Anchored` would be nice.

But that's one piece of what I alluded to earlier with it being easier to show it in code and use it in the next change set.